core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/scheduling/ScheduleGenerator.scala

```diff
@@ -228,4 +224,25 @@ abstract class ScheduleGenerator(
       .addOperator(matReaderPhysicalOp)
       .addLink(readerToDestLink)
   }
+
+  def updatesRegionsWithOutputPortStorage(
+      outputPortsToMaterialize: Set[GlobalPortIdentity],
+      regionDAG: DirectedAcyclicGraph[Region, RegionLink]
+  ): Unit = {
```

Comment:
can you return a new RegionDAG instead of modifying the input regionDAG? for the sake of immutability.

Reply:
We are using this same way of updating the region DAG in all the other existing methods (like `populateDependeeLinks`).